flink-connectors/flink-connector-hive/src/main/java/org/apache/flink/table/catalog/hive/GenericHiveMetastoreCatalog.java

```diff
@@ -229,31 +231,97 @@ public void alterDatabase(String name, CatalogDatabase newDatabase, boolean igno
 	@Override
 	public void dropTable(ObjectPath tablePath, boolean ignoreIfNotExists)
 			throws TableNotExistException, CatalogException {
-		throw new UnsupportedOperationException();
+		try {
+			client.dropTable(
+				tablePath.getDatabaseName(),
+				tablePath.getObjectName(),
+				// Indicate whether associated data should be deleted.
+				// Set to 'true' for now because Flink tables shouldn't have data in Hive. Can be changed later if necessary
```

Comment:
I think what we can do later is to expose this in the API, something like dropTable(ObjectPath, boolean, boolean deleteData).

Reply:
agree, along with 'purge'